litestar/contrib/mongodb_motor/repository/_async.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from litestar.contrib.mongodb_motor.repository._util import wrap_pymongo_exception
+from litestar.contrib.mongodb_motor.repository.types import AsyncMotorCollection, Document
+from litestar.contrib.repository import AbstractAsyncRepository, FilterTypes, RepositoryError
+from litestar.contrib.repository.filters import BeforeAfter, CollectionFilter, SearchFilter
+
+if TYPE_CHECKING:
+    from collections import abc
+    from datetime import datetime
+
+    from pymongo.client_session import ClientSession
```

Comment:
pymongo can be used for sync code, but for async there is need for motor. You can use the types from there, no?